Makefile

```diff
@@ -257,8 +257,17 @@ CFLAGS += -Wno-cast-function-type
 # Defined Symbol (MACROS)
 CFLAGS += -D__HEAP_SIZE=0
 CFLAGS += -DCONFIG_GPIO_AS_PINRESET
-CFLAGS += -DCONFIG_NFCT_PINS_AS_GPIOS
-CFLAGS += -DSOFTDEVICE_PRESENT
+
+# Skip defining CONFIG_NFCT_PINS_AS_GPIOS if the device uses the NFCT.
+ifneq ($(USES_NFCT),yes)
+  CFLAGS += -DCONFIG_NFCT_PINS_AS_GPIOS
+endif
+
+# Skip defining SOFTDEVICE_PRESENT if the device doesn't use a soft device.
+ifneq ($(NO_SOFT_DEVICE),yes)
+  CFLAGS += -DSOFTDEVICE_PRESENT

```

Comment:
To completely remove the SD, we should also exclude all the sd headers file as well to prevent hardfault in run time. Since the current code try its best to detect actual SD existed rather than relying on this macro https://github.com/adafruit/Adafruit_nRF52_Bootloader/blob/master/lib/sdk11/components/libraries/bootloader_dfu/dfu_types.h#L38

This can help to avoid calling sd API if not actually flashed, but a mistake can accidentally invoke an API, therefore a full header excluding is needed.

It is best to use the current `SD_NAME` variable to do this, by default it is as follow, 
https://github.com/adafruit/Adafruit_nRF52_Bootloader/blob/master/Makefile#L91

you should change the `SD_NAME = s140` to `SD_NAME ?= s140`. And having board.mk to define it to `SD=` (null), then modify the include path and macro along with it accordingly.

Reply:
This is a much better solution. Will push a commit in a bit for this, but it will take me a short bit as there's a few places this touches!

Reply:
I actually see that makerdiary did some work to disable softdevice in their branch too, so I'm looking at their commits now to see the sorts of changes I'm needing to make. I am going to remove this one from this PR for the time being, and open a separate PR for that, as it seems it should be a change on it's own unrelated to this device!